cli: cobra TimeFlag type

justinrubek opened the pull request:
Provides a cobra flag type that will enforce that a provided value is an RFC3339 timestamp

justinrubek committed:
feat(cli): cobra TimeFlag type

dskatz reviewed (commented):
Since this is specifically implementing a pflag interface; it should include a unit test that verifies that the interface is correct.

Otherwise it is not clear why Type() and Time() are needed.

justinrubek committed:
test: test time flags

justinrubek committed:
fix: preserve nanoseconds

justinrubek commented:
Good call. I've added tests that uses pflag to parse a TimeFlag from a string. Time() is a utility method that helps get around that this type has an internal `time.Time`, so I've added separate tests for it
